jupyterhub/schema.yaml

```
@@ -212,6 +212,89 @@ properties:
           the `--values` or `-f` flag. During merging, lists are replaced while
           dictionaries are updated.
           ```
+      extraFiles: &extraFiles
+        type: string
+        description: |
+          A dictionary with extra files that should be injected into the pod's
+          container on startup.
+
+          ```yaml
+          hub:
+            extraFiles:
+              <implicit file name>:
+                # name is an optional way to override the implicit file name
+                # that must be unique.
+                name: <explicit file name>
+
+                # mode is by default 0644 and you can optionally override it
+                # either by octal notation (example: 0400) or decimal notation
+                # (example: 256).
+                mode: <file system permissions>
+
+                # mountDir is required and must be the directory you want to
+                # mount the file.
+                mountDir: <directory path>
+
+                # Choose one out of the three ways to represent the actual file
+                # content: data, stringData, or binaryData.
+                #
+                # Both stringData and binaryData should be set to a string
+                # representing the content, where binaryData should be the
+                # base64 encoding of the actual file content.
```

```suggestion
                # stringData must be a string representing the content.
                # binaryData must be the file content encoded as base64.
```

Do we need to worry about character encodings? Or it it assumed to be UTF8?

---

Hmm... Not sure about this in general and doesn't clearly see the steps where there could arise encoding issues.

- Can you pass different encodings of string Helm through values.yaml?
  ?
- Does YAML have some defaults in general?
  It seems UTF-8 is favored by default: https://stackoverflow.com/a/648335
  
Hmmm unless further advised towards a concrete action I'll drop the ball on this consideration. I think we'll be fine.